### PR TITLE
python3.pkgs.django-allauth-2fa: init at 0.11.1

### PR DESCRIPTION
**Pull request by DerDennisOP**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/django-allauth-2fa/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    django-otp
+    qrcode
+  ];
+
+  pythonImportsCheck = [
+    "allauth_2fa"
+  ];
+
+  meta = with lib; {
+    description = "django-allauth-2fa adds two-factor authentication to django-allauth";
```
Comment (natsukium) [inline pkgs/development/python-modules/django-allauth-2fa/default.nix:43]:
```suggestion
    description = "Two-factor authentication for django-allauth";
```
> For brevity, don’t repeat the name of package — just describe what it does.

https://nixos.org/manual/nixpkgs/unstable/#var-meta-description

File: pkgs/development/python-modules/django-allauth-2fa/default.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "valohai";
+    repo = "django-allauth-2fa";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-bm2RwhvX2nfhYs74MM0iZl9U2gHgm0lLlh2tuRRcGso=";
+  };
+
+
```
Comment (natsukium) [inline pkgs/development/python-modules/django-allauth-2fa/default.nix:26]:
```suggestion
```

File: pkgs/development/python-modules/django-allauth-2fa/default.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+, fetchFromGitHub
+, pythonOlder
+, qrcode
+, hatchling
+}:
+
+buildPythonPackage rec {
+  pname = "django-allauth-2fa";
+  version = "0.11.1";
+  format = "pyproject";
```
Comment (natsukium) [inline pkgs/development/python-modules/django-allauth-2fa/default.nix:15]:
```suggestion
  pyproject = true;
```

Comment (DerDennisOP) [reply pkgs/development/python-modules/django-allauth-2fa/default.nix:15]: This does not work for me smh

Comment (natsukium) [reply pkgs/development/python-modules/django-allauth-2fa/default.nix]: The base branch seems to be out of date.
It should be able to build by rebasing to the master branch.

Comment (DerDennisOP) [reply pkgs/development/python-modules/django-allauth-2fa/default.nix]: Well ok, but you can merge it it seams to work anyway.

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 252442` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python310Packages.django-allauth-2fa</li>
    <li>python310Packages.django-allauth-2fa.dist</li>
    <li>python311Packages.django-allauth-2fa</li>
    <li>python311Packages.django-allauth-2fa.dist</li>
  </ul>
</details>

**Comment by DerDennisOP** [comment]
It builds for me, if I remove the "pyproject" var and add format = "pyproject"

**Comment by natsukium** [comment]
Unfortunately, the base branch is too old to build after the merge.

**Comment by SuperSandro2000** [comment]
Denis, you can rebase locally by doing ``git rebase upstream/master``. You can add the remote with ``git remote add upstream https://github.com/nixos/nixpkgs.git `` and ``git fetch --all``,

**Comment by DerDennisOP** [comment]
should work now

**Comment by natsukium** [comment]
@ofborg build python311Packages.django-allauth-2fa

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 252442` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.django-allauth-2fa</li>
    <li>python310Packages.django-allauth-2fa.dist</li>
    <li>python311Packages.django-allauth-2fa</li>
    <li>python311Packages.django-allauth-2fa.dist</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
LGTM, thanks.

**Pull request merged**